Installation instructions for HDF5 (#1787)
ExtremeFLOW/neko: doc/pages/user-guide/installation.md

## Building from source

> To build Neko, you will need a Fortran compiler supporting the Fortran-08 standard, autotools, libtool, pkg-config, a working MPI installation supporting the Fortran 2008 bindings (`mpi_f08`), BLAS/LAPACK and JSON-Fortran. Optional dependencies are PFunit, gslib and ParMETIS.

Reviewer: ```suggestion
To build Neko, you will need a Fortran compiler supporting the Fortran-08 standard, autotools, libtool, pkg-config, a working MPI installation supporting the Fortran 2008 bindings (`mpi_f08`), BLAS/LAPACK and JSON-Fortran. Optional dependencies are PFunit, gslib, HDF5 and ParMETIS.
```

export LD_LIBRARY_PATH=/path/to/installation/lib:$LD_LIBRARY_PATH
```

#### Building HDF5 (optional, but highly recommended)

Download and compile at least version 1.14 of HDF5, with Fortran and MPI
support. Note that you may need to adjust the compilers depending on you machine
(for example, on Cray supercomputers `mpifort` is usually replaced with `ftn`,
and so on).

```shell
wget https://github.com/HDFGroup/hdf5/archive/refs/tags/hdf5_1.14.6.tar.gz
tar xvf hdf5_1.14.6.tar.gz
cd hdf5-hdf5_1.14.6/

> mkdir build && cd build
> 
> cmake -DCMAKE_INSTALL_PREFIX=/path/to/installation -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx -DCMAKE_Fortran_COMPILER=mpifort -DHDF5_ENABLE_PARALLEL=ON -DHDF5_BUILD_FORTRAN=ON ..
> 
> make install

Reviewer 2: How about using modern cmake notation instead
```suggestion
cmake -B build -S ./ --install-prefix /path/to/installation \
    -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx \
    -DCMAKE_Fortran_COMPILER=mpifort -DHDF5_ENABLE_PARALLEL=ON \
    -DHDF5_BUILD_FORTRAN=ON
cmake --build build/ --parallel
cmake --install build/
```

> 

Reviewer: Looks good! If we want to trim down the instructions, the `HL_FORTRAN` parts can be omitted

export LD_LIBRARY_PATH=/path/to/installation/lib:$LD_LIBRARY_PATH
```

#### Building HDF5 (optional, but highly recommended)

Download and compile at least version 1.14 of HDF5, with Fortran and MPI
support. Note that you may need to adjust the compilers depending on you machine
(for example, on Cray supercomputers `mpifort` is usually replaced with `ftn`,
and so on).

```shell
wget https://github.com/HDFGroup/hdf5/archive/refs/tags/hdf5_1.14.6.tar.gz
tar xvf hdf5_1.14.6.tar.gz
cd hdf5-hdf5_1.14.6/
cmake -B build -S ./ --install-prefix /path/to/installation \
    -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx \
    -DCMAKE_Fortran_COMPILER=mpifort -DHDF5_ENABLE_PARALLEL=ON \
    -DHDF5_BUILD_FORTRAN=ON
cmake --build build/ --parallel

> cmake --install build/
> cmake --install build/

Reviewer 2: Double install

```suggestion
cmake --install build/
```
